interfaces/udev/spec.go

```diff
@@ -73,10 +81,12 @@ func udevTag(securityTag string) string {
 // TagDevice adds an app/hook specific udev tag to devices described by the
 // snippet and adds an app/hook-specific RUN rule for hotplugging.
 func (spec *Specification) TagDevice(snippet string) {
-	for _, securityTag := range spec.securityTags {
-		tag := udevTag(securityTag)
-		spec.addEntry(fmt.Sprintf("# %s\n%s, TAG+=\"%s\"", spec.iface, snippet, tag), tag)
-		spec.addEntry(fmt.Sprintf("TAG==\"%s\", RUN+=\"/usr/lib/snapd/snap-device-helper $env{ACTION} %s $devpath $major:$minor\"", tag, tag), tag)
+	if !spec.controlsDeviceCgroup {
```

Comment:
this check is too early because at this point interfaces can be considered in any order so the flag might not be set yet though it's relevant,  we need to consider the flag  in the backend deriveContent or so and not at the point we are inside repo.SnapSpecification.   The way supressHomeIx works in the apparmor backend can give some inspiration.

Reply:
I can certainly move it to there, but this does complicate testing as `deriveContent` would be the function of the udev backend to test, but it's not currently exported, so building up all the required infrastructure to test that the `Setup` function which calls `deriveContent` seemed excessive and complex to me, when all it's really testing is that the original `spec.Snippets` method returns nothing. This is why I chose to implement this logic in `spec.Snippets` instead, as it's easier to test.

Note that technically the result is the same whether we do the check in `deriveContent` or in `Snippets`.

Reply:
I'm a bit confused, this bit of code is not in Snippets at all, it's too early as I said, we can try to catch up still today about this

Reply:
Ah sorry I read the location of this comment wrong. This check is both in `TagDevice` and in `Snippets`. It's redundant to have it both places, but I thought it would be more efficient to have it here in `TagDevice` in addition to `Snippets` as that minimizes the strings we store using `addEntry`, but it's minimal savings so it's probably not a big deal to have it here as well.

Reply:
As I tried to say the issue with doing this only here is that is potentially not correct, it produces a behiavor that depends on the order interfaces are considered.

Reply:
I will remove it here, but what about the equivalent check in `Snippets`?